src/helpers.c

```diff
@@ -29,3 +31,25 @@ int get_model(void)
 
 	return ret;
 }
+
+int chosen_read_u32(const char *name, uint32_t *value)
+{
+    char     path[256];
+    FILE    *fp;
+    uint32_t be_val;
+    size_t   n;
+
+    snprintf(path, sizeof(path), "/sys/firmware/devicetree/base/chosen/%s", name);
+
+    fp = fopen(path, "rb");
+    if (!fp)
+        return errno;
+    n = fread(&be_val, 1, sizeof(be_val), fp);
+    fclose(fp);
+
+    if (n != sizeof(be_val))
+        return EIO;
```

Comment:
Something to keep in mind in the future, when returning an error code we probably want to follow along with how `errno` handles it. e.g. return `-1` and then set `errno` appropriately for consistency. Either that, or, as the kernel does it, return a negative code e.g. `-EIO`